airbyte_cdk/sources/declarative/concurrent_declarative_source.py

```diff
@@ -215,7 +215,7 @@ def _group_streams(
 
                 partition_router_component_definition = (
                     name_to_stream_mapping[declarative_stream.name]
-                    .get("retriever")
+                    .get("retriever", {})
                     .get("partition_router")
```

Comment:
Another case where .get() will return "None" if not present, which doesn't have "partition_router". My fix, allows it to fail gracefully. Alternatively, we could use the same method as above:

```suggestion
                    name_to_stream_mapping[declarative_stream.name]["retriever"]
                    .get("partition_router")
```